lib/std/c/openbsd.zig

```diff
@@ -45,6 +45,52 @@ pub extern "c" fn unveil(path: ?[*:0]const u8, permissions: ?[*:0]const u8) c_in
 pub extern "c" fn pthread_set_name_np(thread: std.c.pthread_t, name: [*:0]const u8) void;
 pub extern "c" fn pthread_get_name_np(thread: std.c.pthread_t, name: [*:0]u8, len: usize) void;
 
+pub const login_cap_t = extern struct {
+    lc_class: [*:0]const u8,
+    lc_cap: [*:0]const u8,
+    lc_style: [*:0]const u8
+};
+
+pub extern "c" fn login_getclass(class: ?[*:0]const u8) ?*login_cap_t;
+pub extern "c" fn login_getstyle(lc: *login_cap_t, style: ?[*:0]const u8, atype: ?[*:0]const u8) ?[*:0]const u8;
+
+pub const auth_session_t = opaque {};
+
+pub extern "c" fn auth_userokay(name: [*:0]const u8, style: ?[*:0]const u8, arg_type: ?[*:0]const u8, password: ?[*:0]const u8) c_int;
+pub extern "c" fn auth_approval(as: ?*auth_session_t, ?*login_cap_t, name: ?[*:0]const u8, type: ?[*:0]const u8) c_int;
+
+pub extern "c" fn auth_userchallenge(name: [*:0]const u8, style: ?[*:0]const u8, arg_type: ?[*:0]const u8, chappengep: *?[*:0]const u8) ?*auth_session_t;
+pub extern "c" fn auth_userresponse(as: *auth_session_t, response: [*:0]const u8, more: c_int) c_int;
+pub extern "c" fn auth_usercheck(name: [*:0]const u8, style: ?[*:0]const u8, arg_type: ?[*:0]const u8, password: ?[*:0]const u8) ?*auth_session_t;
+
+pub extern "c" fn auth_open() ?*auth_session_t;
+pub extern "c" fn auth_close(as: *auth_session_t) c_int;
+
+pub extern "c" fn auth_mkvalue(value: [*:0]const u8) ?[*:0]const u8;
+pub extern "c" fn auth_cat(file: [*:0]const u8) c_int;
+pub extern "c" fn auth_checknologin(lc: *login_cap_t) void;
+
+pub extern "c" fn auth_setdata(as: *auth_session_t, ptr: *anyopaque, len: usize) c_int;
+pub extern "c" fn auth_setitem(as: *auth_session_t, item: auth_item_t, value: [*:0]const u8) c_int;
+pub extern "c" fn auth_getitem(as: *auth_session_t, item: auth_item_t) ?[*:0]const u8;
+pub extern "c" fn auth_setoption(as: *auth_session_t, n: [*:0]const u8, v: [*:0]const u8) c_int;
+pub extern "c" fn auth_setstate(as: *auth_session_t, s: c_int) void;
+pub extern "c" fn auth_getstate(as: *auth_session_t) c_int;
+pub extern "c" fn auth_clean(as: *auth_session_t) void;
+pub extern "c" fn auth_clrenv(as: *auth_session_t) void;
+pub extern "c" fn auth_clroption(as: *auth_session_t, option: [*:0]const u8) void;
+pub extern "c" fn auth_clroptions(as: *auth_session_t) void;
+pub extern "c" fn auth_setenv(as: *auth_session_t) void;
+pub extern "c" fn auth_getvalue(as: *auth_session_t, what: [*:0]const u8) ?[*:0]const u8;
+pub extern "c" fn auth_verify(as: ?*auth_session_t, style: ?[*:0]const u8, name: ?[*:0]const u8, ...) ?*auth_session_t;
+pub extern "c" fn auth_call(as: *auth_session_t, path: [*:0]const u8, ...) c_int;
+pub extern "c" fn auth_challenge(as: *auth_session_t) [*:0]const u8;
+pub extern "c" fn auth_check_expire(as: *auth_session_t) i64;
+pub extern "c" fn auth_check_change(as: *auth_session_t) i64;
+// TODO auth_getpwd requires passwd struct
```

Comment:
`struct passwd;` in Zig is `const passwd = opaque{};`.

Reply:
I just finished filling out the passwd struct and passwd-related functions. Should I replace the `extern struct` with an `opaque`?

Reply:
It looks to be a part of the public API so it should be fine.